#1 - Add java folder for sources

wpmp opened the pull request:
- add java folder in src/main/
- move java sources into the new src/main/java/ folder
- update pom.xml to use correct java version for building the project
- add target/ folder to gitignore

wpmp committed:
#1 - Add java folder for sources

- add java folder in src/main/
- move java sources into the new src/main/java/ folder
- update pom.xml to use correct java version for building the project
- add target/ folder to gitignore

reckart commented on pom.xml:
```diff
@@ -26,4 +26,19 @@
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.0</version>
+                <configuration>
+                    <source>1.8</source>
```
I'd recommend configuring the Java version (and source file encoding) via a properties section from where other plugins may also pick it up:

```
  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <maven.compiler.target>1.8</maven.compiler.target>
    <maven.compiler.source>1.8</maven.compiler.source>
</properties>
```

wpmp committed:
#1 - Add java folder for sources

- configure java version and file encoding in pom.xml with properties instead of compiler plugin
